crates/storage/src/bloom.rs

```diff
@@ -218,6 +216,84 @@ impl AggregateBloom {
     }
 }
 
+impl std::fmt::Debug for AggregateBloom {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use std::hash::{DefaultHasher, Hash, Hasher};
+
+        let mut hasher = DefaultHasher::new();
+        self.bitmap.hash(&mut hasher);
+        let bitmap_hash = hasher.finish();
+
+        f.debug_struct("AggregateBloom")
+            .field("from_block", &self.from_block)
+            .field("to_block", &self.to_block)
+            .field("bitmap_hash", &format!("{:#x}", bitmap_hash))
+            .finish()
+    }
+}
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+struct CacheKey {
+    from_block: BlockNumber,
+    to_block: BlockNumber,
+}
+
+pub(crate) struct AggregateBloomCache(Mutex<SizedCache<CacheKey, AggregateBloom>>);
+
+impl AggregateBloomCache {
+    pub fn with_size(size: usize) -> Self {
+        Self(Mutex::new(SizedCache::with_size(size)))
+    }
+
+    pub fn reset(&self) {
+        self.0.lock().unwrap().cache_reset();
+    }
+
+    pub fn get_many(&self, from_block: BlockNumber, to_block: BlockNumber) -> Vec<AggregateBloom> {
+        let mut cache = self.0.lock().unwrap();
+
+        let from_block = from_block.get();
+        let to_block = to_block.get();
+
+        // Align to the nearest lower multiple of BLOCK_RANGE_LEN.
+        let from_block_aligned = from_block - from_block % AggregateBloom::BLOCK_RANGE_LEN;
+        // Align to the nearest higher multiple of BLOCK_RANGE_LEN, then subtract 1
+        // (zero based indexing).
+        let to_block_aligned = to_block + AggregateBloom::BLOCK_RANGE_LEN
+            - (to_block % AggregateBloom::BLOCK_RANGE_LEN)
+            - 1;
+
+        (from_block_aligned..=to_block_aligned)
+            .step_by(AggregateBloom::BLOCK_RANGE_LEN as usize)
+            .map(|from| {
+                let to = from + AggregateBloom::BLOCK_RANGE_LEN - 1;
+                (
+                    BlockNumber::new_or_panic(from),
+                    BlockNumber::new_or_panic(to),
+                )
+            })
+            .filter_map(|(from_block, to_block)| {
+                let k = CacheKey {
+                    from_block,
+                    to_block,
+                };
+                cache.cache_get(&k).cloned()
```

Comment:
I'm a bit concerned of this cloning here. With the default config the cache can contain ~1GiB of filters, so we'd just clone 1GiB memory here. Even if we're not considering the actual loading of filters missing from the cache and the memory required for that that's 1GiB per running `starknet_getEvents` method. I think that's too  much: we can potentially have 128 concurrent RPC methods running (with the default config).

Maybe we should find a way to avoid actually copying the `AggregateBloom` instances (or at least the bitmap in there)?